lib/futurism/channel.rb

```diff
@@ -16,18 +16,20 @@ def subscribed
     end
 
     def receive(data)
-      resources = data.fetch_values("signed_params", "sgids", "signed_controllers") { |_key| Array.new(data["signed_params"].length, nil) }.transpose
+      resources = data.fetch_values("signed_params", "sgids", "signed_controllers", "urls") { |_key| Array.new(data["signed_params"].length, nil) }.transpose
 
-      resources.each do |signed_params, sgid, signed_controller|
+      resources.each do |signed_params, sgid, signed_controller, url|
         selector = "[data-signed-params='#{signed_params}']"
         selector << "[data-sgid='#{sgid}']" if sgid.present?
 
-        controller_lookup = ControllerLookup.from(signed_string: signed_controller)
-        controller_lookup.setup_env!(connection: connection)
-        controller = controller_lookup.controller
+        controller = Resolver::Controller.from(signed_string: signed_controller)
+        renderer = Resolver::Controller::Renderer.for(controller: controller,
+                                                      connection: connection,
+                                                      url: url,
+                                                      params: @params)
```

Comment:
where dose that ivar come from? I can't seem to find it

Reply:
This already exists from `ActionCable::Channel::Base` see https://github.com/rails/rails/blob/fe76a95b0d252a2d7c25e69498b720c96b243ea2/actioncable/lib/action_cable/channel/base.rb#L144

In my testing, it's not anything very interesting (basically the controller and action values), but I figured I'd include those base params because it's possible there are interesting things there the consumer will want.

Reply:
Ah of course, thanks for clarifying. I’d maybe add a comment for my future self 😬